src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const fromParts = fromFormat.slice(0, -1);
+  const toParts = toFormat.slice(0, -1);
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateObj = {};
+
+  fromParts.forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
+
+  const formattedDate = toParts
+    .map((part) => {
+      if (part === 'YYYY' && dateObj['YY']) {
+        const year = parseInt(dateObj['YY'], 10);
+
+        if (year < 30) {
+          return '20' + dateObj['YY'];
+        } else {
+          return '19' + dateObj['YY'];
+        }
+      } else if (part === 'YY' && dateObj['YYYY']) {
+        return dateObj['YYYY'].slice(-2);
+      } else {
+        return dateObj[part];
+      }
+    })
+    .join(toSeparator);
+
+  return formattedDate;
```

Comment:
The function does not currently handle the case where the 'fromFormat' and 'toFormat' arrays contain different sets of date parts (e.g., 'fromFormat' has 'DD' but 'toFormat' does not). Ensure that the function can handle such cases gracefully, either by returning an error or by documenting this behavior as a limitation.